pkg/matcher/annotation_tasks_install_test.go

```diff
@@ -423,6 +429,66 @@ func TestGetPipelineFromAnnotations(t *testing.T) {
 			},
 			wantErr: "only one pipeline is allowed on remote",
 		},
+		{
+			name: "test-annotations-unknown-hub",
+			annotations: map[string]string{
+				keys.Pipeline: "[foo://bar]",
+			},
+			wantLog: "custom catalog foo is not found",
+			wantErr: "could not get remote pipeline \"foo://bar\": returning empty",
+		},
+		{
+			name:            "test-get-from-custom-hub",
+			gotPipelineName: "pipeline",
+			annotations: map[string]string{
+				keys.Pipeline: "[anotherHub://chmouzie]",
+			},
+			wantLog: "successfully fetched pipeline chmouzie from custom catalog HUB anotherHub on URL https://mybelovedhub",
+			remoteURLS: map[string]map[string]string{
+				testHubURL + "/resource/" + testCatalogHubName + "/pipeline/chmouzie": {
+					"body": `{"data": {"LatestVersion": {"version": "0.1"}}}`,
+					"code": "200",
+				},
+				fmt.Sprintf("%s/resource/%s/pipeline/chmouzie/0.1/raw", testHubURL, testCatalogHubName): {
+					"body": readTDfile(t, "pipeline-good"),
+					"code": "200",
+				},
+			},
+		},
+		{
+			name:            "test-get-from-hub-latest",
+			gotPipelineName: "pipeline",
+			annotations: map[string]string{
+				keys.Pipeline: "[chmouzie]",
+			},
+			remoteURLS: map[string]map[string]string{
+				testHubURL + "/resource/" + testCatalogHubName + "/pipeline/chmouzie": {
+					"body": `{"data": {"LatestVersion": {"version": "0.1"}}}`,
+					"code": "200",
+				},
+				fmt.Sprintf("%s/resource/%s/pipeline/chmouzie/0.1/raw", testHubURL, testCatalogHubName): {
+					"body": readTDfile(t, "pipeline-good"),
+					"code": "200",
+				},
+			},
+		},
+		{
+			name:            "test-get-from-hub-specific-version",
+			gotPipelineName: "pipeline",
+			annotations: map[string]string{
+				keys.Pipeline: "[chmouzie:0.2]",
+			},
+			remoteURLS: map[string]map[string]string{
+				testHubURL + "/resource/" + testCatalogHubName + "/pipeline/chmouzie/0.2": {
+					"body": `{}`,
+					"code": "200",
+				},
+				fmt.Sprintf("%s/resource/%s/pipeline/chmouzie/0.2/raw", testHubURL, testCatalogHubName): {
+					"body": readTDfile(t, "pipeline-good"),
+					"code": "200",
+				},
+			},
+		},
```

Comment:
Can we also add a failure scenario to have multiple annotation for pipeline ?
I mean
```
annotations: map[string]string{
				keys.Pipeline: "[chmouzie:0.2]",
                                keys.Pipeline: "[chmouzie:0.1]",
			},
```

Reply:
Its a map, you cannot specify same key twice

Reply:
sorry i mean 
```
annotations: map[string]string{
				keys.Pipeline: "[chmouzie:0.2]",
                                keys.Pipeline-1: "[chmouzie:0.1]",
			},
```
the negative scenario

Reply:
This is an annotation which is not parsed by code, so it will not affect anything, we don't check that these annotations should not exist, so it will not fail. It will act like any other random annotation we can do. Code only looks for `pipeline`

Reply:
okay got it